hamster_gtk/preferences/preferences_dialog.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+
+
+# This file is part of 'hamster-gtk'.
+#
+# 'hamster-gtk' is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# 'hamster-gtk' is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with 'hamster-gtk'.  If not, see <http://www.gnu.org/licenses/>.
+
+
+"""This module provides the preferences dialog."""
+
+
+from __future__ import absolute_import, unicode_literals
+
+import collections
+from gettext import gettext as _
+
+import hamster_lib
+from gi.repository import GObject, Gtk
+
+from hamster_gtk.preferences.widgets import (ComboFileChooser,
+                                             HamsterComboBoxText,
+                                             HamsterSpinButton,
+                                             SimpleAdjustment, TimeEntry)
+
+
+class PreferencesDialog(Gtk.Dialog):
+    """A dialog that shows and allows editing of config settings."""
+
+    def __init__(self, parent, app, initial, *args, **kwargs):
+        """
+        Instantiate dialog.
+
+        Args:
+            parent (Gtk.Window): Dialog parent.
+            app (HamsterGTK): Main app instance. Needed in order to retrieve
+                and manipulate config values.
+        """
+        super(PreferencesDialog, self).__init__(*args, **kwargs)
+
+        self._parent = parent
+        self._app = app
+
+        self.set_transient_for(self._parent)
```

Comment:
No, it is already used (up the stream](https://github.com/jtojnar/hamster-gtk/blob/b5c3b886f76a602b12c0e0d2b3e439071f747ca0/hamster_gtk/hamster_gtk.py#L101).